src/1-theory/10-async.adoc

```diff
@@ -0,0 +1,407 @@
+= Asynchronous Paradigms
+
+So far the discussion of these paradigms has probably seemed rather similar,
+they are all following a request and response structure. Getting a response from
+a request is one way to do things, but there are lots of situations where a
+client might expect more than one response, or a response with more information
+later as more work happens on the server side.
+
+// First, some terminology!
+
+// **Synchronous** -
+// **Asynchronous** -
+// **Concurrency** -
+// **Parallelagram** -
+
+// push v pull
+
+== Client-Server "Real-Time" Models
+
+Let's have a quick run through of some of the common models for doing
+asynchronous stuff. Even thought you might not want to use some of these, it is
+good to know what they are, even if that is only so you can recommend not doing
+it when somebody else brings them up in a planning meeting.
+
+=== Short Polling
+
+The first approach that often gets mentioned is short polling, which is the
+client-server equivalent of a child in the backseat of the car asking "are we
+there yet? are we there yet? are we there yet?". The client is making the same
+GET request to the same resource every few seconds, waiting for some sort of
+change in the answer.
+
+[quote,sp3c1,https://stackoverflow.com/questions/4642598/short-polling-vs-long-polling-for-real-time-web-applications]
+----
+00:00:00 C-> Is the cake ready?
+00:00:01 S-> No, wait.
+00:00:01 C-> Is the cake ready?
+00:00:02 S-> No, wait.
+00:00:02 C-> Is the cake ready?
+00:00:03 S-> Yeah. Have some lad.
+00:00:03 C-> Is the other cake ready? ...
+----
+
+The only benefit of short polling is that it is _very_ easy to implement. The
+client can just write a while loop that makes a request, checks for a response,
+then sleeps for a few seconds before making another request.
+
+The main downside with short polling is that it's a very chatty approach, which
+makes it awful to scale. The more clients you have making these requests the
+more load your API has to handle, and it can grow exponentially. Conditional
+network caching can help a bit here, but even then it sucks the battery life on
+mobile devices.
+
+Generally short polling is rubbish, and you should try to avoid it.
+
+=== Long Polling
+
+A slight tweak on the concept of short polling, long polling maintains an open
+connection with the server until the answer comes in.
+
+[quote,sp3c1,https://stackoverflow.com/questions/4642598/short-polling-vs-long-polling-for-real-time-web-applications]
+----
+00:00:00 C-> Is the cake ready?
+00:00:03 S-> Yeah. Have some lad.
```

Comment:
```suggestion
* some time later *
00:00:03 S-> Yeah. Have some.
```

Reply:
Well after reading the next sentence I noticed the timestamps... anyhow... might make this easier to understand at a glance, but feel free to ignore me.

Reply:
had the same thought initially but did not mention it, :+1:  for making this more clear.